apps/web/prisma/schema.prisma

```diff
@@ -278,3 +280,23 @@ model Campaign {
 
   team Team @relation(fields: [teamId], references: [id], onDelete: Cascade)
 }
+
+model ApiLog {
+  id          String   @id @default(cuid())
+  teamId      Int
+  method      String
+  path        String
+  status      Int
+  duration    Int // in milliseconds
+  apiKeyId    Int?
+  requestBody Json? // Store request body
+  response    Json? // Store response data
+  error       String? // Store error message if any
+  createdAt   DateTime @default(now())
+
+  team   Team    @relation(fields: [teamId], references: [id])
```

Comment:
recently i'm leaning into not using foreign key for use cases like this.

one such article: https://www.shayon.dev/post/2023/355/do-you-really-need-foreign-keys/

let's add indexes and remove the relation. we can handle it in application logic.

Reply:
Emulating foreign key in application layer and not in database layer could give some benefits like the blog post above mentioned. But It also brings some issues on the table. Implementing `onCascade:Delete` on application layer feels problematic in my personal view. 

Also the example shown on the blog post, is for deleting the entries. So equivalent operation is our case would be deleting the `team` should also delete the `api-logs` of those `team`, right? So, now the main question is how likely the team is going to be deleted which will execute this inefficient query.

Also, Since all other tables in the schema were using the foreign key approach. Having some tables with foreign key and some without will be a headache for maintenance. What do you think?